File: py/125.py
# Author: Andrew Hamara

# Solution to leetcode problem 125. Valid Palindrom

class Solution:
    def isPalindrome(self, s:str) -> bool:
        wordAsList = [x.upper() for x in s if x.isalnum()]
        return wordAsList == wordAsList[::-1]

File: go/125.go
// Author: Andrew Hamara

// Solution to leetcode problem 125. Valid Palindrome

import (
    "regexp"
)

func isPalindrome(s string) bool {
    reg := regexp.MustCompile("[^A-Za-z0-9]+")
    alnum := strings.ToLower(reg.ReplaceAllString(s, ""))

    left := 0
    right := len(alnum) - 1

    for right > left {
        if alnum[left] != alnum[right] {
            return false
        }
        left += 1
        right -= 1
    }
    return true

}
